provider/resources.go

```diff
@@ -7004,5 +7004,6 @@ func Provider() tfbridge.ProviderInfo {
 	// Add a CSharp-specific override for aws_s3_bucket.bucket.
 	prov.Resources["aws_s3_bucket_legacy"].Fields["bucket"].CSharpName = "BucketName"
 
+	shimv2.SetInstanceStateStrategy(prov.P.ResourcesMap().Get("aws_wafv2_web_acl"), shimv2.CtyInstanceState)
```

Comment:
This enables the fix selectively for aws_wafv2_web_acl.

Reply:
Is there any guidance/docs around when to use `CtyInstanceState` and its pros and cons?

Reply:
https://github.com/pulumi/pulumi-terraform-bridge/pull/887

Reply:
OK but what about going forward: a comment on this line explaining why we're using it here? A section in the bridged provider docs explaining when to use it when upgrading a provider?

Reply:
> explaining when to use it when upgrading a provider

I don't have any guidance here. The new behavior should be the default behavior but isn't currently due to limitations of our current test capabilities. It's rolling out for aws_wafv2_web_acl selectively ahead of time to fix an important issue.

>  a comment on this line explaining why we're using it here
I'm a fan of git-blame for this but can add a comment.